### PR TITLE
Remove unused iscsi functions

### PR DESCRIPTION
**Pull request by qiankehan**
Remove the functions of setting target acls, set  initiator name of iscsiadm and restarting iscsid.

### CODE REVIEW

**Comment by qiankehan** [comment]
@chunfuwen @dzhengfy @lijinlijin Please help to review it. The RHEL8.8 virtual disk test run \#29 has passed with this PR

**Comment by chunfuwen** [comment]
@qiankehan  can you give more details why remove above functions?
In terms of iscsi, there are multiple jobs ,which use it in various ways.
We need carefully every small step to change iscsi from previous my experience where it cost lots of time for debugging even though small change is made on iscsi.

**Comment by qiankehan** [comment]
> @chunfuwen @dzhengfy @lijinlijin Please help to review it. The RHEL8.8 virtual disk test run `#29` has passed with this PR

1. 51723fcbfb4b6332ec6ff3f55317b493e6c58b13: From the search result of avocado-vt, tp-libvirt, tp-qemu and tp-spice, the function `set_chap_acls_target` and `delete_chap_account` are unused:
```
➜  Software ag '(set_chap_acls_target|delete_chap_account)' avocado-vt tp-libvirt tp-qemu tp-spice 
avocado-vt/selftests/unit/test_iscsi.py
141:    def setup_stubs_delete_chap_account(self, iscsi_obj):

avocado-vt/virttest/iscsi.py
451:    def delete_chap_account(self):
631:    def set_chap_acls_target(self):
```
2. After the former commit,  [targetcli ACLs](https://github.com/open-iscsi/targetcli-fb/blob/master/targetcli.8#L182) is not used. ACLs are to controll what iscsi initiator could connect the iscsi target. If they are not used, all initiator name could connect to the target. So we don't have to change the initiator name.
3. Since the former commit, we don't change the initiator name. We don't have to restart iscsid. `iscsiadm` will manage all the things of iscsi. And the iscsid could be started by the socket when iscsiadm connect to it[1]. 

[1]: https://github.com/open-iscsi/open-iscsi/commit/5d0e19fcc1cea77a72647cf96c5d3d773e8ee277

I have run the most of single host iscsi cases on iscsi-initiator-utils-6.2.1.4-4 qemu-kvm-6.2.0-27 libvirt-8.0.0-12, none of them fails due to iscsi:
```
for i in $(grep iscsi cases |grep -vE '(migrat|multi)');do avocado run --vt-type libvirt --vt-machine-type q35 $i;done
```
The failed cases are caused by:
- Fail to create qed image
- Fail to set sys_rawio capability

**Comment by qiankehan** [comment]
[result.log](https://github.com/avocado-framework/avocado-vt/files/10264268/result.log)
The test result

**Comment by qiankehan** [comment]
> restart_iscsid ACL initiatior Those functions are very natural features for iscsi, so even if we don't use them currently, I don't suggest we need remove them

What is the exact meaning of **natural**? Did you mean it is open-iscsi related?

In my opinion, to keep it or remove a function/class, just need to check if it is being or will be used.
I don't see any possibility that they will be used in tp-libvirt or tp-qemu. What's more, I looked up the commits and found they are never used:
```
➜  tp-libvirt git:(master) PAGER=cat; git log -1 --format="%h    %s"
daaeb66ae    Merge pull request #4764 from xiaodwan/add_kubevirt_support

➜  tp-libvirt git:(master) PAGER=cat; for i in $(git log --format='%h');do git show $i|grep -qiE "(set_chap_acls_target|delete_chap_account|restart_iscsid)"; if [ $? -eq 0 ];then git log $i -1 --format="%h    %s";fi;done

➜  tp-qemu git:(master) PAGER=cat; git log -1 --format="%h    %s"
13725995    Merge pull request #3606 from 6-dehan/new-api-cpustats

➜  tp-qemu git:(master) PAGER=cat; for i in $(git log --format='%h');do git show $i|grep -qiE "(set_chap_acls_target|delete_chap_account|restart_iscsid)"; if [ $? -eq 0 ];then git log $i -1 --format="%h    %s";fi;done

```

**Comment by chunfuwen** [comment]
I just throw two cents here about my personal overview about this PR.
1)I don't see clear positive impact this PR will bring in
2)Look like this PR remove 3 methods to reduce maintenance effort, but if we look at virttest/iscsi.py commit history, we almost has no change for this file after it is merged
3)If we check tp-libvirt code only, there are 500+ places(egrep -nr 'iscsi' --include=*.py |wc -l) ,which is related to usage of iscsi. And iscsi.py are widely used in various ways. In order to make sure the change will not bring regression, lots of additional test efforts are needed,and plus maintainers's efforts for reviewing
4) In the past, there was one ever tiny change in iscsi.py, during debugging, even finding that avocado run will be hung under special condition, which we may believe it is not related iscsi.py.  So that actually happened.

**Comment by qiankehan** [comment]
> I just throw two cents here about my personal overview about this PR. 1)I don't see clear positive impact this PR will bring in 2)Look like this PR remove 3 methods to reduce maintenance effort, but if we look at virttest/iscsi.py commit history, we almost has no change for this file after it is merged 3)If we check tp-libvirt code only, there are 500+ places(egrep -nr 'iscsi' --include=*.py |wc -l) ,which is related to usage of iscsi. And iscsi.py are widely used in various ways. In order to make sure the change will not bring regression, lots of additional test efforts are needed,and plus maintainers's efforts for reviewing 4) In the past, there was one ever tiny change in iscsi.py, during debugging, even finding that avocado run will be hung under special condition, which we may believe it is not related iscsi.py. So that actually happened.

1. the "clear positive impact" is **keep it more simple and clear**
2. Do you mean the **virttest/iscsi.py** is too old to change? I don't think it is a good idea for an active project
3. Sure. This PR is [widely tested](https://github.com/avocado-framework/avocado-vt/pull/3580#issuecomment-1358759626) as well. If any iscsi cases are missing, you can list them, and then I will run the tests.
4. Even though **iscsi.py** is too complex to change, we should not refuse to change. I know your concerns about the possible regressions to master branch. How about creating a develop branch and merging it to that branch if no questions on the codes and test themselfves? This develop branch could be used for code refactoring like this PR. When the develop branch pass the full tests, we can merge it to master.

**Comment by chunfuwen** [comment]
@clebergnu could you give your thoughts about this PR?

File: virttest/iscsi.py
Comment (chunfuwen) [inline virttest/iscsi.py:835]: How can you make sure restart_iscsid() here is not necessary?

Comment (qiankehan) [reply virttest/iscsi.py:835]: > How can you make sure restart_iscsid() here is not necessary?

The caller function **export_target** sets up the iscsi target, as well as the server side, while the iscsid is the initiator  , as well as the client side. Since they don't establish a connection before **export_target**, the client side iscsid doesn't need to restart. Just like you don't have to restart your browser for a new website.

**Review by chunfuwen — APPROVED** [review]
LGTM

**Comment by dzhengfy** [comment]
@qiankehan I added some descriptions about your test results like you said below:
Tested with some cases in virtual disk/scsi_hostdev/backingchain/storage features. No regression.
` (1/1) type_specific.io-github-autotest-libvirt.virtual_disks.multivms.coldplug.vms_sharable_test.block_type: PASS (50.96 s)`

Based on above results, I am confident it is low risk to merge this one. Thanks.

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**
